### PR TITLE
Add variable sized data blob erasure coding.

### LINKED ISSUE #195: Variable sized data erasure coding

### PR DESCRIPTION
**Pull request by greywolve**
Closes #195

### CODE REVIEW

**Review by carlos-romano — APPROVED** [review]

File: internal/erasurecoding/erasurecoding_test.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+			if err != nil {
+				t.Fatalf("Decode failed: %v", err)
+			}
+
+			require.Equal(t, data, dataOut)
+		})
+	}
+}
+
+func TestEncodeDecodeInitialTestVector(t *testing.T) {
```
Comment (bamzedev) [inline internal/erasurecoding/erasurecoding_test.go:96]: Should we move this test to the integration folder to be consistent with the rest that tests test vectors? Also saw that we have the `readsolomon_test` also testing vectors.

Comment (greywolve) [reply internal/erasurecoding/erasurecoding_test.go:96]: Do you feel like this is really an integration test if it's only testing one component?

We also only have one semi official test vector at the moment, maybe that can be moved to integration when the proper erasure coding vectors land?

Comment (greywolve) [reply internal/erasurecoding/erasurecoding_test.go:96]: I could have probably just inlined that initial test vector

Comment (bamzedev) [reply internal/erasurecoding/erasurecoding_test.go:96]: Imo if it is a test vector it should live with the other test vector cases. If you inline it you hardcode the test vector, right? It's called integration test, but that is just an incorrect naming. In reality it is a test vector folder.

Comment (greywolve) [reply internal/erasurecoding/erasurecoding_test.go:96]: This isn't an official test vector technically, I'll remove the vectors folder and just put the json in the package. When the official vectors land then I'll add those tests in the integration tests folder.

Comment (greywolve) [reply internal/erasurecoding/erasurecoding_test.go:96]: (basically I also want a test like this at the package level)

**Review by bamzedev — APPROVED** [review]

**Pull request merged**